taichi/ir/basic_stmt_visitor.cpp

```diff
@@ -8,9 +8,12 @@ BasicStmtVisitor::BasicStmtVisitor() {
 }
 
 void BasicStmtVisitor::visit(Block *stmt_list) {
-  for (auto &stmt : stmt_list->statements) {
+  std::vector<Stmt *> statements;
+  // Make a copy in case the pass modifies the block itself
```

Comment:
Maybe `DelayedIRModifier` will be unnecessary after this PR?

Reply:
No, that's still needed. `DelayedIRModifier` guarantees that IR is unmodified during a pass. That's a stronger property than what we have here.